src/lib/__tests__/crunchInterceptor.test.js

```diff
@@ -0,0 +1,60 @@
+import express from "express"
+import graphqlHTTP from "express-graphql"
+import request from "supertest"
+import bodyParser from "body-parser"
+import { makeExecutableSchema, addMockFunctionsToSchema } from "graphql-tools"
+import { crunch } from "graphql-crunch"
+
+import crunchInterceptor from "../crunchInterceptor"
+
+describe("crunchInterceptor", () => {
+  let app
+
+  beforeEach(() => {
+    const schema = makeExecutableSchema({
+      typeDefs: `
+        type Query {
+          greeting: String
+        }
+      `,
+    })
+    addMockFunctionsToSchema({ schema })
+    app = express()
+    app.use(
+      "/",
+      bodyParser.json(),
+      crunchInterceptor,
+      graphqlHTTP({
+        schema,
+        graphiql: false,
+      })
+    )
+  })
+
+  it("should pass the result through unchanged when no param is present", () => {
+    return request(app)
+      .get("/?query={greeting}")
+      .set("Accept", "application/json")
+      .expect(res => {
+        expect(res.body.data).toMatchObject({ greeting: "Hello World" })
+      })
+  })
+
+  it("should crunch the result when param is present", () => {
+    return request(app)
+      .get("/?query={greeting}&crunch")
+      .set("Accept", "application/json")
+      .expect(res => {
+        expect(res.body.data).toMatchObject(crunch({ greeting: "Hello World" }))
+      })
+  })
+
+  it("should not crunch an introspection query", () => {
+    return request(app)
+      .get("/?query={__schema{types{name}}}&crunch")
+      .set("Accept", "application/json")
+      .expect(res => {
+        expect(Array.isArray(res.body.data)).toBeFalsy()
```

Comment:
The expectation is that a crunched result will be an array whereas a non-crunched result will be an object. This isn't a very sophisticated test, but it at least provides some level of validation. I'm not convinced this should even be a thing though... more on that below.